Add imenu and refactor regexes

jrogov opened the pull request.

jrogov committed:
Add imenu and refactor regexes

jrogov committed:
fix false positives

False Positives:
1. } else = status {
2. "string": { # set start

psibi commented:
@arikai Thanks, can you attach a snapshot of the before and the after part with this PR ?

I should really add tests for these some day! :-)

jrogov committed:
enforce rule-head font-locking

Example:
"ip_replacement_forbidden[party_id] = false {" was not properly colored due to assignment (or literal, not really sure)
